service/geopmdpy/system_files.py

```diff
@@ -682,7 +685,21 @@ def remove_batch_server(self, client_pid):
         self.check_client_active(client_pid, 'remove_batch_server')
         self._sessions[client_pid].pop('batch_server')
         self._update_session_file(client_pid)
-        # TODO: Unlink any shared memory files left around
+
```

Comment:
Please add some tests.  This code should not do any pattern matching.  The file paths should be deterministic based on the batch server PID.  As it it written now this could delete files that are unrelated to the particular batch server that ended.